### PR TITLE
Fix instrumented runtime for glibc < 2.17

### PR DESCRIPTION
**Pull request by bobot**
- Compile the instrumented runtime, for catching errors in CAML_INSTR
    macros.
- Compile the debug runtime for compile time errors in Assert.

### CODE REVIEW

File: configure [hunk 1]
```diff
@@ -1061,6 +1061,18 @@ if test "$with_curses" = "yes"; then
...
 
+# For instrumented runtime
+# (clock_gettime need -lrt for glibc before 2.17)
+if test "$with_instrumented_runtime" = "true"; then
+  for libs in "" "-lrt"; do
+    if sh ./hasgot $libs clock_gettime; then
+      inf "clock_gettime functions found (with libraries '$libs')"
+      instrumented_runtime_libs="${libs}"
+      break
+    fi
```
Comment (mshinwell) [inline configure:1074]: Should this refuse to build the instrumented runtime if clock_gettime is in fact absent on the system?

File: configure [hunk 2]
```diff
@@ -1061,6 +1061,18 @@ if test "$with_curses" = "yes"; then
   done
 fi
 
+# For instrumented runtime
+# (clock_gettime need -lrt for glibc before 2.17)
```
Comment (mshinwell) [inline configure]: "needs"

**Comment by mshinwell** [comment]
I made a couple of comments, but this otherwise looks OK.
I agree this should go in 4.03 since it's needed for the testsuite.

**Comment by mshinwell** [comment]
(The Travis failure seems to be camlp4 only due to a constructor having been renamed.  @diml is working on fixing that.)

File: configure [hunk 3]
```diff
@@ -1061,6 +1061,23 @@ if test "$with_curses" = "yes"; then
   done
 fi
 
+# For instrumented runtime
+# (clock_gettime needs -lrt for glibc before 2.17)
+if test "$with_instrumented_runtime" = "true"; then
```
Comment (damiendoligez) [inline configure]: Please! change this to

```
if $with_instrumented_runtime; then
```

File: configure [hunk 4]
```diff
@@ -1061,6 +1061,23 @@ if test "$with_curses" = "yes"; then
...
+  with_instrumented_runtime=false #enabled it only if found
+  for libs in "" "-lrt"; do
+    if sh ./hasgot $libs clock_gettime; then
+      inf "clock_gettime functions found (with libraries '$libs')"
+      instrumented_runtime_libs="${libs}"
+      with_instrumented_runtime=true;
+      break
+    fi
+  done
+  if test "$with_instrumented_runtime" = "false"; then
```
Comment (damiendoligez) [inline configure]: Likewise, use

```
if ! $with_instrumented_runtime; then
```

**Comment by damiendoligez** [comment]
OK for merging.

**Comment by mshinwell** [comment]
@bobot Changes entry perhaps?

**Comment by damiendoligez** [comment]
I don't think a Changes entry is warranted, as instrumented runtime is a new feature.

**Pull request merged**